inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/region_yolo_gpu_ref.cl

```diff
@@ -12,94 +12,95 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "include/common.cl"
-#include "include/data_types.cl"
+#include "include/include_all.cl"
 
+#if INPUT0_LAYOUT_BFYX
 #define IW INPUT0_SIZES[0]
 #define IH INPUT0_SIZES[1]
 #define IC INPUT0_SIZES[2]
 #define IB INPUT0_SIZES[3]
+#elif INPUT0_LAYOUT_BYXF
+#define IC INPUT0_SIZES[0]
+#define IW INPUT0_SIZES[1]
+#define IH INPUT0_SIZES[2]
+#define IB INPUT0_SIZES[3]
+#endif
 
 inline UNIT_TYPE FUNC(logistic_activate)(UNIT_TYPE x) {
     return 1. / (1. + exp(-x));
 }
 
-inline int FUNC(entry_index)(int width, int height, int coords, int classes,
-                       int outputs, int batch, int location,
-                       int entry) {
-    int n = location / (width * height);
-    int loc = location % (width * height);
-    return batch * outputs + n * width * height * (coords + classes + 1) +
-        entry * width * height + loc;
-}
+inline int FUNC(output_index)(int batch, int region_num, int xy, int feature_offset) {
+    int region_offset = region_num * (COORDS + CLASSES + 1);
 
 #if DO_SOFTMAX
-inline void FUNC(softmax_generic)(const __global UNIT_TYPE* src_data, __global UNIT_TYPE* dst_data,
-                            int B, int C, int W, int H, int i)
-{
-    for (int b = 0; b < B; b++) {
-        UNIT_TYPE max = src_data[b*C*H*W + i];
-        for (int c = 0; c < C; c++) {
-            UNIT_TYPE val = src_data[b*C*H*W + c*H*W + i];
-            if (val > max) max = val;
-        }
-
-        UNIT_TYPE expSum = 0;
-        for (int c = 0; c < C; c++) {
-            dst_data[b*C*H*W + c*H*W + i] = exp(src_data[b*C*H*W + c*H*W + i] - max);
-            expSum += dst_data[b*C*H*W + c*H*W + i];
-        }
-
-        for (int c = 0; c < C; c++) {
-            dst_data[b*C*H*W + c*H*W + i] = dst_data[b*C*H*W + c*H*W + i] / expSum;
-        }
-    }
-}
+    return OUTPUT_GET_INDEX(batch, (feature_offset + region_offset) * INPUT0_SIZE_X * INPUT0_SIZE_Y + xy, 1, 1);
+#else
+    int x_index = xy % INPUT0_SIZE_X;
+    int y_index = (xy / INPUT0_SIZE_X) % (INPUT0_SIZE_Y);
```

Comment:
`x_index` and `y_index` are already calculated out of kernel. You can just reuse it, I think..